maps-db/src/main/resources/db/migration/V1.00.01__create_map_index_table.sql

```diff
@@ -0,0 +1,33 @@
+comment on database maps_db is 'Database used to store, track and serve map information';
+
+create table map (
+  id serial primary key,
+  name varchar(256) not null unique check(length(name) > 3),
+  date_created timestamptz not null default now(),
+  date_updated timestamptz not null default now()
+);
+
+create table category (
+  id serial primary key,
+  name varchar(32) unique not null check (length(name) > 2),
+  date_created timestamptz not null default now()
+);
+
+insert into category (name) values
+  ('BEST'),
+  ('GOOD'),
+  ('EXPERIMENTAL');
+
+
+create table map_version (
+  id serial primary key,
+  map_id integer not null references map(id),
+  version integer not null check (version > 0),
```

Comment:
Sooner or later we might want to revisit this too

Reply:
I'm pretty sure we do not want semver. Checksumming the map and then making map versions immutable would solve all semver problems. Simply then if a game client is missing the map that is uniquely identified, it can download it and you then have a new map version on your system.

Reply:
w.r.t map semver, how is a map maker to know what is a compatible map change and what is not? We _do not_ want map makers to have to have to continue to be 'mini-developers' knowing implementation details of how the code works. Currently they must know far too much about the internals that it just creates a really high barrier to entry. If I change for example a unit stat and upload a new version, the game engine should just do the right thing without me having to ask a dev if that is semver version increase, and to which number. WE then get into further issues with how the engine should handle such semver version numbers, it just is overly complex I think

Reply:
I just read your post on the forum where you explained a couple of things.
I was more thinking about making use of SQL to determine the latest version.
There's no need to increment a version id if we could just look at the "creation date" and/or the compatible engine versions to determine a suited version to download.

Reply:
I'm not convinced that the game engine semver is appropriate when it comes to map compatibility. For example, if we add a new property to maps and deprecate an old one, are we then making a major version update? 

I'm pretty sure we will likely be moving map parsing logic into a sub-project that is below game-headed. At that point the map parsing engine is no longer aware of the 'product.version' at all. 

Last, game engine version does not capture the range of compatibility for a map. We essentially could have a case where  map is compatible with 2.3, 2.4, 3.0, but not 3.1.

We should also consider that there can be multiple versions of a map suitable for a game engine, but perhaps we don't want the latest. The map itself could be updated to use latest properties (for example a map that does work with 2.0 by using legacy properties but is then updated to use 2.0 specific properties). 

I'm suspecting that we'll eventually create a "map version", IE: a specification value, similar to "HTML 4.0" that specifies the valid properties that can be defined in the map. Today we re-use `version` like crazy, which I think is over-use of that concept.

The version number here is in part needed just so we can transition from `triplea_maps.yaml` and have similar functionality. The game clients should be able to read data from either the server or the `triplea_maps.yaml` while we transition.